lib/ansible/modules/cloud/amazon/ec2_vpc_subnet_facts.py

```diff
@@ -77,74 +81,82 @@
     subnet_ids: "{{ subnet_facts.results|map(attribute='subnets.0.id')|list }}"
 '''
 
-try:
-    import boto.vpc
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
-except ImportError:
-    HAS_BOTO = False
-
+import traceback
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
-
-
-def get_subnet_info(subnet):
-
-    subnet_info = {'id': subnet.id,
-                   'availability_zone': subnet.availability_zone,
-                   'available_ip_address_count': subnet.available_ip_address_count,
-                   'cidr_block': subnet.cidr_block,
-                   'default_for_az': subnet.defaultForAz,
-                   'map_public_ip_on_launch': subnet.mapPublicIpOnLaunch,
-                   'state': subnet.state,
-                   'tags': subnet.tags,
-                   'vpc_id': subnet.vpc_id}
-
-    return subnet_info
-
-
-def list_ec2_vpc_subnets(connection, module):
-
-    filters = module.params.get("filters")
-    subnet_dict_array = []
-
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    AWSRetry,
+    HAS_BOTO3,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
+from botocore.exceptions import ClientError
+
+
+@AWSRetry.backoff()
+def describe_subnets(module, client, subnet_ids, filters):
+    """
+    Describe Subnets
+
+    module  : AnsibleModule object
+    client  : boto3 client connection object
+    subnet_ids : A list of subnet ids to get
+    filters : A complex dict of filters to apply to the results
+    """
+
+    # init empty list for return vars
+    subnet_info = list()
+
+    # Get the basic VPC info
     try:
-        all_subnets = connection.get_all_subnets(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
+        response = client.describe_subnets(SubnetIds=subnet_ids, Filters=filters)
+    except ClientError as err:
+        module.fail_json(msg=err.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(err.response))
 
-    for subnet in all_subnets:
-        subnet_dict_array.append(get_subnet_info(subnet))
+    for subnet in response['Subnets']:
+        # for backwards compatibility
+        subnet['id'] = subnet['SubnetId']
+        # convert tag list to ansible dict
+        subnet['Tags'] = boto3_tag_list_to_ansible_dict(subnet['Tags'])
+        subnet_info.append(camel_dict_to_snake_dict(subnet))
```

Comment:
you need to do camel_dict_to_snake_dict before you do the tag conversion (otherwise tag names get converted to lower case which is unexpected)